source/common/http/path_utility.cc

```diff
@@ -64,10 +64,11 @@ void PathUtil::mergeSlashes(RequestHeaderMap& headers) {
   if (path.find("//") == absl::string_view::npos) {
     return;
   }
-  const absl::string_view prefix = absl::StartsWith(path, "/") ? "/" : absl::string_view();
-  const absl::string_view suffix = absl::EndsWith(path, "/") ? "/" : absl::string_view();
-  headers.setPath(absl::StrCat(
-      prefix, absl::StrJoin(absl::StrSplit(path, '/', absl::SkipEmpty()), "/"), query, suffix));
+  const absl::string_view path_prefix = absl::StartsWith(path, "/") ? "/" : absl::string_view();
+  const absl::string_view path_suffix = absl::EndsWith(path, "/") ? "/" : absl::string_view();
+  headers.setPath(absl::StrCat(path_prefix,
+                               absl::StrJoin(absl::StrSplit(path, '/', absl::SkipEmpty()), "/"),
+                               path_suffix, query));
```

Comment:
TBH this transform scares me from a security perspective, but I guess it's off by default so someone has to consciously enable it.

@htuch for visibility and @envoyproxy/senior-maintainers pass

At a glance it looks fine.

Reply:
Yeah, it seems this could be a source of potential vulnerability if enabled. @euroelessar do you think this is worth a security advisory? This is all tempered by this not being RFC mandated behavior.